### ISSUE TITLE
Work out how to do a secure delete in Python

### ISSUE THREAD

**Issue by teor2345** [privcount/privcount#14]
SecureCounters.detach_blinding_shares says:
TODO: secure delete

I think we could overwrite the values with zeroes. That might be a start.
Or is there a pythonic way of doing this?

**Comment by teor2345** [comment]
The python cryptography module (which privcount uses) says there is no reliable secure memory wiping API in python:

> Memory wiping is used to protect secret data or key material from attackers with access to uninitialized memory. This can be either because the attacker has some kind of local user access or because of how other software uses uninitialized memory.
> 
> Python exposes no API for us to implement this reliably and as such almost all software in Python is potentially vulnerable to this attack. The CERT secure coding guidelines assesses this issue as “Severity: medium, Likelihood: unlikely, Remediation Cost: expensive to repair” and we do not consider this a high risk for most users.

https://cryptography.io/en/latest/limitations/#lack-of-secure-memory-wiping

Unless we can reliably do a secure wipe of memory, I think we should follow the cryptography module example and document it as a limitation. (Creating our own implementation would not give any guarantees, and might lead users to believe we have solved this problem.)

**Comment by robgjansen** [comment]
I agree. Documenting the limitation seems OK to me given the circumstance.

**Comment by robgjansen** [comment]
I think we need to securely erase both the blinding values, and also the noise values. If you can recover the blinding value, then it can be used to learn about the counter actual value before the end of the collection phase. If you can recover the noise value, then you can learn about the actual value of the counter even after the collection phase ends.

**Comment by teor2345** [comment]
We might be able to do this with ctypes, but it will be messy and potentially unreliable.

**Comment by teor2345** [comment]
Or, we could use Cython, and get a speed boost as well.
https://github.com/cython/cython/wiki/FAQ
